### DISCUSSION TITLE
Android signing issue

### DISCUSSION THREAD

**Discussion by MurphysMagic** [codemagic-ci-cd/codemagic-docs#1611] [Q&A] [answered]
I'm getting the following error whilst signing my Android app:

Execution failed for task ':app:signReleaseBundle'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.FinalizeBundleTask$BundleToolRunnable
   > Failed to read key ******** from store "/Users/builder/.keystores/android_keystore_oyrup7_7.jks": Get Key failed: Given final block not properly padded. Such issues can arise if a bad key is used during decryption.

I'm not sure where the issue lies, is it because of a corrupt key or incorrect password or something else?

**Comment by dtrdic** [comment]
Hello there,
Can you please share your build Id?

**Reply by MurphysMagic** [reply]
Hello it's this: 6451023eea0bc87b57a8ab0b

**Reply by dtrdic** [reply]
hey @MurphysMagic,how do you generate your .jks file ? Did you run the command as suggested in the doc [here](https://docs.codemagic.io/yaml-code-signing/signing-android/#generating-a-keystore)  ? The error is kind of saying that the uploaded key or its provided keys are not correct. Let's re-generate the key and make sure that its password is correct.

Looking forward to your feedback.

**Reply by MurphysMagic** [reply]
Hello,

I've got an existing app on the store signed with this key so would that affect me submitting a new build with this re-generated key?

**Reply by dtrdic** [reply]
Ok i got the impression that this was your first build. Yes, submitting a new build of your app with a regenerated signing key will affect your existing app on the store because the Google Play Store recognizes the signing key used to sign your app as a unique identifier of your app. If you submit a new build of your app with a different signing key, it will be treated as a new app and you will not be able to update your existing app with it. So let's try like this first;
run this command locally to make sure you are using the correct passwords/alias
`keytool -list -v -storepass "my-store-pass" -keypass "my-key-pass" -alias "my-alias" -keystore "my-keystore.jks"`

**Reply by MurphysMagic** [reply, marked as answer]
Hello, I managed to re-add the keystore to Codemagic with the password I had on record and it seems like I must have made a mistake the first time as it works now, thanks for your help :)

**Reply by dtrdic** [reply]
I'm glad to hear it:) Cheers
